astrogrism/HST/dispersion_models.py

```diff
@@ -30,23 +30,57 @@ def __init__(self, ematrix, offset, inv=False):
         self._model_set_axis = False
 
     # Note that in the inverse case, input "t" here is actually dx or dy
-    def evaluate(self, x, y, t):
+    def evaluate(self, x, y, t, t_op = "multiply"):
 
         e = self.ematrix
         offset = self.offset
-        if isinstance(x, np.ndarray):
-            if len(x) == 1:
-                x = float(x)
-            else:
-                raise ValueError(f"x is array: {x}")
-        if isinstance(y, np.ndarray):
-            if len(y) == 1:
-                y = float(y)
-            else:
-                raise ValueError(f"y is array: {y}")
+        reshape_output = False
+
+        # Handle reshaping of x and y to handle arrays if needed
+        if x.ndim != y.ndim:
+            raise ValueError("Input x and y arrays must have same dimensionality."
+                             "2D arrays will be used as-is, 1D arrays will be broadcast"
+                             "together. See documentation for further detail.")
+
+        if x.ndim == 2:
+            if x.shape != y.shape:
+                raise ValueError("If x and y inputs are 2D their shapes must match")
+
+        # If x and y are 1D arrays, assume we need to create a meshgrid
+        elif x.ndim == 1 and x.shape != (1,) and y.shape != (1,):
+            #mesh = np.meshgrid(x, y)
+            #x = mesh[0]
+            #y = mesh[1]
+            if x.shape != y.shape:
+                raise ValueError("If x and y are 1D arrays they must have the"
+                                 "same shape. See documentation for instructions"
+                                 "for dispersing a 2D region.")
+
```

Comment:
Yup, no meshgrid, which means if x and y are arrays they have to have the same number of elements. You can't give it a 4 element x array and an 8 element y array, for example. I think the case on line 39 can actually be dropped since we're not supporting 2D array inputs.